Source/Core/Core/PowerPC/JitArm64/JitArm64_SystemRegisters.cpp

```diff
@@ -247,41 +247,55 @@ void JitArm64::mfspr(UGeckoInstruction inst)
   case SPR_TL:
   case SPR_TU:
   {
-    // The inline implementation here is inaccurate and out of date as of PR3601
-    FALLBACK_IF(true);  // Fallback to interpreted version.
+    ARM64Reg Wg = gpr.GetReg();
+    ARM64Reg Xg = EncodeRegTo64(Wg);
+
+    ARM64Reg Wresult = gpr.GetReg();
+    ARM64Reg Xresult = EncodeRegTo64(Wresult);
 
-    /*
     ARM64Reg WA = gpr.GetReg();
     ARM64Reg WB = gpr.GetReg();
     ARM64Reg XA = EncodeRegTo64(WA);
     ARM64Reg XB = EncodeRegTo64(WB);
 
+    ARM64Reg VC = fpr.GetReg();
+    ARM64Reg VD = fpr.GetReg();
+    ARM64Reg SC = EncodeRegToSingle(VC);
+    ARM64Reg SD = EncodeRegToSingle(VD);
+
     // An inline implementation of CoreTiming::GetFakeTimeBase, since in timer-heavy games the
     // cost of calling out to C for this is actually significant.
-    MOVI2R(XA, (u64)&CoreTiming::g_global_timer);
-    LDR(INDEX_UNSIGNED, XA, XA, 0);
-    MOVI2R(XB, (u64)&CoreTiming::g_fake_TB_start_ticks);
-    LDR(INDEX_UNSIGNED, XB, XB, 0);
-    SUB(XA, XA, XB);
+
+    MOVP2R(Xg, &CoreTiming::g);
+
+    LDR(INDEX_UNSIGNED, WA, PPC_REG, PPCSTATE_OFF(downcount));
+    m_float_emit.SCVTF(SC, WA);
+    m_float_emit.LDR(32, INDEX_UNSIGNED, SD, Xg,
+                     offsetof(CoreTiming::Globals, last_OC_factor_inverted));
+    m_float_emit.FMUL(SC, SC, SD);
+    m_float_emit.FCVTS(Xresult, SC, ROUND_Z);
+
+    LDP(INDEX_SIGNED, XA, XB, Xg, offsetof(CoreTiming::Globals, global_timer));
+    SXTW(XB, WB);
```

Comment:
There is no SXTW instruction, this is an alias for ADD. So what about:
ADD XA, XA, WB, SXTW
SUB Xresult, XA, Xresult